src/Compilers/VisualBasic/Portable/Compilation/MethodCompiler.vb

```diff
@@ -1180,7 +1179,9 @@ Namespace Microsoft.CodeAnalysis.VisualBasic
                 Dim compilation = compilationState.Compilation
                 If compilation.ShouldAddEvent(method) Then
                     If block Is Nothing Then
-                        compilation.SymbolDeclaredEvent(sourceMethod)
+                        If Not method.IsPartial OrElse method.PartialImplementationPart Is Nothing Then
```

Comment:
Could you please add an assert at the beginning of the method that it is never called for a partial definition?